SGM/src/sgmstereo/sgmstereo.cpp

```diff
@@ -52,7 +52,6 @@ SGMStereo::SGMStereo(int _w, int _h, int minDisparity, int maxDisparity, int num
 	}
 
 	float rec_penalty2 = 1.0f / m_penalty2;
-	wLUT = new float[256];
```

Comment:
I see what you're doing here - adding this to the class member. Overall, I'm a fan, but I think that this should go on the heap. That way, if future changes introduce an indexing error, we have less chance of corrupting the stack. Also, it'll take up less space on the stack, so if we decide to create a bunch of these for some reason, we'll be less likely to segfault.

I think an easy way would be to keep the float* in the class, and allocate the buffer in the class constructor.